plasmapy/particles/particle_class.py

```diff
@@ -105,7 +110,23 @@ def _category_errmsg(particle, category: str) -> str:
     return errmsg
 
 
-class Particle:
+class AbstractParticle(ABC):
+    """
+    An abstract base class that defines the interface for particles.
+    """
+
+    @property
+    @abstractmethod
+    def mass(self) -> Union[u.Quantity, Real]:
+        pass
+
+    @property
+    @abstractmethod
+    def charge(self) -> Union[u.Quantity, Real]:
+        pass
```

Comment:
I think I used `particle.name` in `RefactorTracker` (hope you don't mind I call it by branch name, it's simpler for me that way - I've been checking it out a lot lately). Should I switch to `str(particle)`?